rotkehlchen/exchanges/okx.py

```diff
@@ -0,0 +1,486 @@
+        """
+        https://www.okx.com/docs-v5/en/#rest-api-trade-get-order-history-last-3-months
+
+        May raise
+        - RemoteError from _api_query_list_paginated
+        """
+        raw_trades = self._api_query_list_paginated(
+            endpoint='trades',
+            pagination_key='ordId',
+            options={
+                'start_ts': start_ts,
+                'end_ts': end_ts,
+            },
+        )
+
+        trades: list[Trade] = []
+        for raw_trade in raw_trades:
+            trade = self.trade_from_okx(raw_trade)
+            if trade is not None:
+                trades.append(trade)
+
+        return trades, (start_ts, end_ts)
+
+    def query_online_margin_history(
+            self,
+            start_ts: Timestamp,
+            end_ts: Timestamp,
+    ) -> list[MarginPosition]:
+        return []  # noop for okx
+
+    def query_online_deposits_withdrawals(
+            self,
+            start_ts: Timestamp,
+            end_ts: Timestamp,
+    ) -> list[AssetMovement]:
+        """
+        https://www.okx.com/docs-v5/en/#rest-api-funding-get-deposit-history
+        https://www.okx.com/docs-v5/en/#rest-api-funding-get-withdrawal-history
+
+        May raise
+        - RemoteError from _api_query_list_paginated
+        """
+        deposits = self._api_query_list_paginated(
+            endpoint='deposits',
+            pagination_key='ts',
+            options={
+                'start_ts': start_ts,
+                'end_ts': end_ts,
+            },
+        )
+        withdrawals = self._api_query_list_paginated(
+            endpoint='withdrawals',
+            pagination_key='ordId',
+            options={
+                'start_ts': start_ts,
+                'end_ts': end_ts,
+            },
+        )
+
+        movements: list[AssetMovement] = []
+        for raw_movement in deposits:
+            movement = self.asset_movement_from_okx(
+                raw_movement=raw_movement,
+                category=AssetMovementCategory.DEPOSIT,
+            )
+            if movement is not None:
+                movements.append(movement)
+        for raw_movement in withdrawals:
+            movement = self.asset_movement_from_okx(
+                raw_movement=raw_movement,
+                category=AssetMovementCategory.WITHDRAWAL,
+            )
+            if movement is not None:
+                movements.append(movement)
+
+        return movements
+
+    def query_online_income_loss_expense(
+            self,
+            start_ts: Timestamp,
+            end_ts: Timestamp,
+    ) -> list[LedgerAction]:
+        return []  # noop for okx
+
+    def trade_from_okx(self, raw_trade: dict[str, Any]) -> Optional[Trade]:
+        """
+        Converts a raw trade from OKX into a Trade object
+        """
+        try:
+            timestamp = ts_ms_to_sec(TimestampMS(int(raw_trade['cTime'])))
+            base_asset_str, quote_asset_str = raw_trade['instId'].split('-')
```

Comment:
This can raise `ValueError` due to `split()` if the string does not contain `'-``.
```suggestion
            try:
                base_asset_str, quote_asset_str = raw_trade['instId'].split('-')
            except ValueError as e:
                raise DeserializationError(f'Expected pair {raw_trade["instId"]} to contain a "-"') from e
```

And then remove the ValueError handling from below. This is the only thing that can raise ValueError here, right? Errors need to make sense. Otherwise if we see the following in a user's logs we don't easily know what happened: `ValueError: not enough values to unpack (expected 2, got 1)`